docs/shared/orog.yaml

```diff
@@ -0,0 +1,15 @@
+orog:
+  execution:
+    batchargs:
+      cores: 1
+      walltime: 00:05:00
+    executable: /path/to/orog
+    files_to_copy:
+      fort.15: /path/to/fix/thirty.second.antarctic.new.bin
+      landcover30.fixed: /path/to/fix/landcover30.fixed
+      fort.235: /path/to/fix/gmted2010.30sec.int
+    grid_file: /path/to/netcdf/grid/file
+    rundir: /path/to/run/dir
+platform:
+  account: me
+  scheduler: slurm
```

Comment:
This will also need to be multiple grid files, but that's in the case of global grid support. Regional is the one that only needs one -- tile 7 of the "cube".

I think `orog-gsl` also needs some work in this area. If I'm [reading it correctly](https://github.com/ufs-community/uwtools/blob/6273086aa4c0a64dd5a6d5b3a97f31d23a13f322/src/uwtools/drivers/orog_gsl.py#L23-L34) that one only handles one `input_grid_file` in its current state. Perhaps we could add some backlog work to upgrade them both to multiple grids for global support?

Reply:
I think you're right that `OrogGSL` currently only supports a single input grid file. But that driver appears to expect to be told

- The pathname of the file, whatever it might be called, and then
- The resolution, tile number, and halo number, from which it _constructs_ the filename as it appears in the run directory

My memory is hazy but IIRC the upstream task that generates these input grid files does not (necessarily?) create files with `C%s_grid.tile%s.halo%s.nc` names, but `orog_gsl` requires that format, so the driver takes care of linking an arbitrarily-named file to the correct filename in the run directory. I don't think that driver is actually being used anywhere yet, so maybe there are design issues.

I find it unfortunate if we have to expect users to know magic names like `fort.235`, `C%s_grid.tile%s.halo%s.nc`, etc., that are _required by the component_ and get those right in their configs.

Reply:
To say it another way, if the component is compiled so that it opens and reads a `fort.235` file, then IMO `fort.235` should never appear in a user config, because it's not configurable. It's an invariant fact about the component that the driver would add value by hiding.

Reply:
I totally agree on the user knowledge about `fort.??` file names! That becomes a chore that the uw can help make more human-readable/understandable.

I'm not sure if in this case we want to break it up over 2+ sections since the `fort.??` files are invariant, but the 3 file has changed names.

I am fine with changing this one to capture only the `files_to_link` section to reduce some of the complexity. Since the ufs_utils example uses links, I think that's a fair way to go. SRW uses copies, but doesn't seem necessary.

Reply:
I'll probably agree with whatever you decide to do with the tradeoffs -- I just wanted to make sure we considered them.
> but the 3 file has changed names
Do you mean the landcover file, where different component versions expect different names? Maybe we have no choice for now but to leave that fully configurable. Ultimately, I feel like we'll have to provide versioned drivers: Maybe a `version:` key in the driver's block, optional with a sensible default, and conditional logic depending on the version. I'm not sure what the _value_ of `version:` would be (maybe a git hash if the component doesn't have its own version number), but then it might link the landcover file into the run directory as either `landcover30.fixed` or `landcover.umd.30s.nc` based on `version:`, if that's the deciding factor.

If/when we implement versioned drivers, it might be cleanest to do so by subclassing, so that we might e.g. `class OrogV2(Orog)` and just override the task responsible for the landcover file, rather than littering the code with `if` statements.

Reply:
Eek, yeah, I think I was going for `but the 3rd file has changed names`, and that is the `landcover` file.

I agree with the versioning comments. I like where that could be going, but dread the day that we get there.